pkg/services/ngalert/schedule/schedule.go

```diff
@@ -424,84 +422,106 @@ func (sch *schedule) ruleEvaluationLoop(ctx context.Context) error {
 }
 
 func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key models.AlertRuleKey, evalCh <-chan *evalContext, stopCh <-chan struct{}) error {
-	sch.log.Debug("alert rule routine started", "key", key)
+	logger := sch.log.New("uid", key.UID, "org", key.OrgID)
+	logger.Debug("alert rule routine started")
 
-	evalRunning := false
-	var attempt int64
-	var alertRule *models.AlertRule
-	for {
-		select {
-		case ctx := <-evalCh:
-			if evalRunning {
-				continue
-			}
+	orgID := fmt.Sprint(key.OrgID)
+	evalTotalMetric := sch.metrics.EvalTotal.WithLabelValues(orgID)
+	evalDurationMetric := sch.metrics.EvalDuration.WithLabelValues(orgID)
+	evalTotalFailuresMetric := sch.metrics.EvalFailures.WithLabelValues(orgID)
```

Comment:
nit: one last thing I missed, it's [very common](https://prometheus.io/docs/practices/naming/#metric-and-label-naming) for counters to end in `total` or `failure` and histograms to use `duration`. I'd drop the `metric` suffix from these variables.